pymeasure/instruments/nf/li5660.py

```diff
@@ -0,0 +1,445 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+from pymeasure.instruments import Instrument #, RangeException
+from pymeasure.instruments.validators import truncated_range, strict_discrete_set
+# from pymeasure.adapters import PrologixAdapter
+# from pymeasure.instruments.validators import truncated_range, strict_discrete_set
+
+# import numpy as np
+# import time
+# from io import BytesIO
+# import re
+
+class LI5660(Instrument):
+
+    def __init__(self, resourceName, **kwargs):
+        super().__init__(
+            resourceName,
+            "NF Lock-In Amplifier LI5600",
+            **kwargs
+        )
+    
+    amplifier_id = Instrument.measurement(
+        "*IDN?", """Reads the instrument identification """
+    )
+
+    # Current Screen display mod
+    display_screen = Instrument.control(
+        ":DISP?", ":DISP %s", 
+        """There are 3 types of measurement screen display: Normal, Large and Fine Measurement screens""",
+        validator = strict_discrete_set,
+        values={"Normail": "NORM", "Large": "LARG", "Fine": "FINE"},
+        map_values=True
+    )
+    ###################
+    # Reference Sigal #
+    ###################
+    signal_termainal = Instrument.control(
+        ":ROUT?", ":ROUT %s",
+        """ A property that control the input connector """,
+        validator=strict_discrete_set,
+        values={"A": "A",
+                "AB": "AB",
+                "C": "C",
+                "Current": "I",
+                "HF": "HF"},
+        map_values=True
+
+    )
+
+    reference_signal_source = Instrument.control(
+        ":ROUT2?", "ROUT2 %s",
+        """ A property that control the reference signal source """,
+        validator=strict_discrete_set,
+        values={"Ref In": "RINP",
+                "Internal": "IOSC",
+                "Signal": "SINP"},
+        map_values=True
+    )
+
+    reference_frequency_source = Instrument.control(
+        ":ROSC:SOUR?",":ROSC:SOUR %s",
+        """ A property that control the reference frequency source for frequency synthesis.""",
+        validator=strict_discrete_set,
+        values={"Internal": "INT", "External": "EXT"},
+        map_values=True
+    )
+
+    ###############
+    # Sensitivity #
+    ###############
+    auto = Instrument.measurement(
+        ":AUTO:ONCE",
+        """ a property that automatically set the sensitivity and time constant once to match the
+        reference signal for the signal being measured at the time. Performs auto setting one time """
+    )
+
+    auto_current = Instrument.control(
+        "CURR:AC:RANG:AUTO?", "CURR:AC:RANG %d",
+        """ A property that continuous automatic selection current sensitivity function.
+        On(enabled), Off(disabled) and Once(auto sets current sensitivity one time).""",
+        values={"On": 1 , "Off": 0, "ONCE": "ONCE"},
```

Comment:
Shouldn't you also include the `validator=strict_discrete_set` option, like you did with the other controls.

Reply:
Thanks, validator added